gossip/src/epoch_slots.rs

```diff
@@ -81,31 +81,44 @@ impl std::convert::From<flate2::DecompressError> for Error {
 }
 
 impl Flate2 {
-    fn deflate(mut unc: Uncompressed) -> Result<Self> {
-        let mut compressed = Vec::with_capacity(unc.slots.block_capacity());
+    fn deflate(unc: Uncompressed) -> Result<Self> {
+        // View into the slots array as raw bytes
+        let bits = unc.slots.as_raw_slice();
+        // Assume compressed version will be smaller than the original
+        let mut compressed = Vec::with_capacity(bits.len());
+        // Perform compression and check status to make sure everything fits
         let mut compressor = Compress::new(Compression::best(), false);
-        let first_slot = unc.first_slot;
-        let num = unc.num;
-        unc.slots.shrink_to_fit();
-        let bits = unc.slots.into_boxed_slice();
-        compressor.compress_vec(&bits, &mut compressed, FlushCompress::Finish)?;
+        let status = compressor.compress_vec(bits, &mut compressed, FlushCompress::Finish)?;
+        if status != flate2::Status::StreamEnd {
+            return Err(Error::CompressError);
+        }
+
         let rv = Self {
-            first_slot,
-            num,
+            first_slot: unc.first_slot,
+            num: unc.num,
             compressed,
         };
-        let _ = rv.inflate()?;
```

Comment:
Apparently this was a hack to validate that compression worked out. A crude but effective one I guess.

Reply:
This seems like it was a sanity check.
I don't know how often it does fail though.
Have you checked if it ever fails or not?

Reply:
can we keep this line as a debug check:
```rust 
debug_assert_matches!(rv.inflate(), Ok(_));
```